packages/kbn-apm-config-loader/src/apm_config.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import type { ServiceConfigDescriptor } from '@kbn/core-base-server-internal';
+
+const migrationSchema = schema.object({
+  apm: schema.object({
+    active: schema.maybe(schema.boolean()),
+    serverUrl: schema.maybe(schema.uri()),
+    secretToken: schema.maybe(schema.string()),
+    globalLabels: schema.object({}, { unknowns: 'allow' }),
+  }),
+});
+
+export type ElasticApmConfigType = TypeOf<typeof migrationSchema>;
+
+export const elasticApmConfig: ServiceConfigDescriptor<ElasticApmConfigType> = {
+  path: 'elastic',
+  schema: migrationSchema,
+};
```

Comment:
Even if atm not used for anything else than the `apm` subpath, I think it's risky to have the whole `elastic` config prefix being registered and 'owned' by this package, as we wouldn't be able to add other non-apm properties to it (e.g if we want to add `elastic.somethingElse` at some point)

I would rather expose the `apm` schema from the package

```ts
export const apmConfigSchema = schema.object({
    active: schema.maybe(schema.boolean()),
    serverUrl: schema.maybe(schema.uri()),
    secretToken: schema.maybe(schema.string()),
    globalLabels: schema.object({}, { unknowns: 'allow' }),
  }),
```

and have the `elasticConfig` (with the `elastic` path) being defined in core, e.g

```ts
export const elasticApmConfig: ServiceConfigDescriptor<ElasticApmConfigType> = {
  path: 'elastic',
  schema: schema.object({
   apm: apmConfigSchema,
  }),
};
```